### PR TITLE
Instructions top padding

### PR DESCRIPTION
**Pull request by Bjvanminnen**
This is a bunch of code change for a very small actual change.

Actual change is to make it so that when we have instructions in our top pane view, there's no top padding
image

Before doing this I made a semi-sizable refactor. It feels to me like our instructions component is really two distinct components (depending on whether we provide markdown or not). I wanted to make that more explicit.

I then afterwards realized that it made more sense for aniGif and authoredHints to still live in the base component, so I moved them back.

### CODE REVIEW

**Comment by Bjvanminnen** [comment]
Validated changes on an applab level, an artist level, a starwars level (with instructions2) and a jigsaw level (with an anigfi)

**Comment by Bjvanminnen** [comment]
Also let me know if you think it's silly to have separated these two components - but I think this feels cleaner to me.

File: apps/src/templates/instructions/MarkdownInstructions.jsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+    var style = styles.standard;
+    if (this.props.inTopPane) {
+      style = styles.inTopPane;
+    } else if (this.props.mardownClassicMargins) {
+      style = styles.classic;
+    }
+
+    return (
+      <div>
+        <div
```
Comment (islemaster) [inline apps/src/templates/instructions/MarkdownInstructions.jsx]: Why the wrapper div here?

File: apps/src/templates/instructions/NonMarkdownInstructions.jsx [hunk 2]
```diff
@@ -0,0 +1,33 @@
+var styles = {
+  main: {
+    marginBottom: 35,
+    marginLeft: 80
+  }
+};
+
+var NonMarkdownInstructions = React.createClass({
```
Comment (islemaster) [inline apps/src/templates/instructions/NonMarkdownInstructions.jsx]: This looks like a good candidate for a functional stateless component.  Actually, I guess both of these do.

File: apps/src/templates/instructions/TopInstructions.jsx [hunk 3]
```diff
@@ -101,7 +101,10 @@ var TopInstructions = React.createClass({
         </div>
         <div style={collapseStyle}>
           <div style={bodyStyle}>
-            <Instructions renderedMarkdown={procesMarkdown(this.props.markdown)}/>
+            <Instructions
+              renderedMarkdown={procesMarkdown(this.props.markdown)}
```
Comment (islemaster) [inline apps/src/templates/instructions/TopInstructions.jsx]: s/proces/process ?

File: apps/src/templates/instructions/MarkdownInstructions.jsx (hunk 1 above)
Comment (Bjvanminnen) [inline apps/src/templates/instructions/MarkdownInstructions.jsx]: Originally had some other stuff in here too, then pulled it back out to the outer instructions. Will get rid of.

File: apps/src/templates/instructions/NonMarkdownInstructions.jsx (hunk 2 above)
Comment (Bjvanminnen) [inline apps/src/templates/instructions/NonMarkdownInstructions.jsx]: I think you're right. Both of these started out more complex, and then got simpler :) In simple forms, I think they fit the bill.

File: apps/src/templates/instructions/MarkdownInstructions.jsx [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    renderedMarkdown: React.PropTypes.string.isRequired,
+    markdownClassicMargins: React.PropTypes.bool,
+    inTopPane: React.PropTypes.bool
+  },
+
+  render: function () {
+    var style = styles.standard;
+    if (this.props.inTopPane) {
+      style = styles.inTopPane;
+    } else if (this.props.mardownClassicMargins) {
```
Comment (islemaster) [inline apps/src/templates/instructions/MarkdownInstructions.jsx]: s/mardown/markdown

**Comment by islemaster** [comment]
Excellent general approach of breaking these out until they're very simple.  This isn't overengineering - it's going _just_ far enough to make reasoning about the parts really easy.  Nice work.

Typos in property names slipping through point to a need for test coverage.  I'd recommend stopping now and getting the test boilerplate for the two stateless components done, and maybe for the `Instructions` parent component as well.

After fixing typos, LGTM, tests recommended.

File: apps/src/templates/instructions/TopInstructions.jsx (hunk 3 above)
Comment (Bjvanminnen) [inline apps/src/templates/instructions/TopInstructions.jsx]: Note that for this particular typo, I was at least consistent (the line at the beginning of the file is `var procesMarkdown = require('markdown');` so tests wouldn't have changed anything (there's no actual bug, just a typo).

The other typo is a legit bug :(

Comment (islemaster) [inline apps/src/templates/instructions/TopInstructions.jsx]: :D That's fair.

**Pull request merged**